recipes/ctre/all/conanfile.py

```diff
@@ -0,0 +1,48 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class CtreConan(ConanFile):
+    name = "ctre"
+    homepage = "https://github.com/hanickadot/compile-time-regular-expressions"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Compile Time Regular Expression for C++17/20"
+    topics = ("cpp17", "regex", "compile-time-regular-expressions")
+    license = "Apache 2.0 with LLVM Exception"
+    no_copy_source = True
+    settings = "compiler"
+
+    _source_name = "compile-time-regular-expressions"
+    _source_subfolder = "source_subfolder"
+
+    def _validate_compiler_settings(self):
+        compiler = self.settings.compiler
+        version = tools.Version(self.settings.compiler.version)
+
+        if compiler == "Visual Studio":
+            if self.settings.compiler.get_safe("cppstd"):
+                tools.check_min_cppstd(self, "17")
+            if version < "15":
+                raise ConanInvalidConfiguration("ctre doesn't support MSVC < 15")
+        elif compiler == "gcc" and version < "7.4":
+            raise ConanInvalidConfiguration("ctre doesn't support gcc < 7.4")
```

Comment:
typically gcc c++17 support is available for 7 and up, is there a reason for 7.4?

Reply:
https://github.com/hanickadot/compile-time-regular-expressions#supported-compilers :man_shrugging:

Reply:
I indeed just went off of what had been mentioned there. It could be bugs in GCC prior to 7.4, missing C++17 features given that lots of compilers declare "C++17 support" and don't mean it (looking at you VS 2017...), or who-knows-what.

Reply:
I added a comment at the top of the function with that link for future reference.